### ISSUE TITLE
Enhancement: Prevent wizards from learning spells from other classes (+ wands)

### ISSUE THREAD

**Issue by chrpow** [Muttley/foundryvtt-shadowdark#566]
Currently, if a wizard holds a scroll of a priest spell, the UI for learning that spell still shows up with no indication that it isn't allowed. Similarly, all characters are able to activate wands regardless of their spellcasting class (or lack thereof).

While some people might want to homebrew and allow PCs to use magic from other classes in certain cases, it would be nice if there was at least some kind of indication that it isn't allowed, especially since it isn't always obvious which classes are allowed to use a certain wand if you don't have the spell lists memorized.

**Comment by Muttley** [comment]
There are a couple of issues here:

> Currently, if a wizard holds a scroll of a priest spell, the UI for learning that spell still shows up with no indication that it isn't allowed.

I think here we should just have an additional "Are You Sure?" warning when you're learning a scroll from outside your own class.  Otherwise we limit future or homebrew classes who *are* allowed to learn any spells, and it's trivial to rectify things if a player makes a mistake.

> Similarly, all characters are able to activate wands regardless of their spellcasting class (or lack thereof).

This was already fixed a while back.  Only classes with a spellcasting ability set can use magic devices.

So to summarise, the first issue we will add an "Are You Sure?" message for spells outside the character's own class.
